### PR TITLE
Ruby on Rails capstone project - Budget Manager

### PR DESCRIPTION
**Pull request by Mutai-Gilbert**
- [ ] In this pull request I was able to implement the following:

- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

Video presentation - https://www.loom.com/share/aac682fa335249e28873f11b49e49b03?sid=961e988c-e2bd-4dac-b7be-f100f90c5300

Live Project:- https://budget-manager-web-service.onrender.com

Test User:-
email: test@gmail.com
pass: test123

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,13 +70,13 @@
```
Comment (Nemwel-Boniface) [inline README.md:71]: While I am not able to highlight all the sections of the README, I will describe the section I mean.

- [x] When I started your project locally I got an error. The error was however solved by running ```rake assets:precompile``` to pre-compile the assets before they were used. KIndly ensure that you include this in your documentation to ease the process of setting up the project locally and using it.

**This is the error I got:**

Screenshot from 2023-07-19 18-04-10

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
##  Hello @Mutai-Gilbert  :wave: 
### This is a manual review
#   :recycle:  Request Changes :recycle: 

- [ ] Good job working on your project so far! However, there are some issues remaining to move to the next step.




**I have to add the changes here as I am not able to highlight them in the "files changed"**

- [x] You did well in the video presentation. However, you seem to not have mentioned the aim/ purpose of the application you built. What is is mainly supposed to be used for? What will it help me achieve? Kindly retake the video demo and in the new video kindly include the aim for building the project.

- [x] Your project seems to not have any mechanism of "data Encapsulation" when you create two accounts and create your records and then login, you will see them even in the new account despite you not creating the records in this new account. This is caused because in your categories index action, you are basically fetching all the records instead of fetching the records that only belong to a specific user. Kindly update the index action to only show data the currently logged-in user alone. 

- [x] In your home page "Categories, I seem to not be able to logout from your application. The only way I found to be able to logout is that I have to click on "Add category" then I will be able to see the logout button. Kindly ensure that this button is included in all the relevant pages of your application including the Categories page.

**This is how I am seeing it right now:**

[Screencast from 19-07-2023 06:58:33 ALASIRI.webm](https://github.com/Mutai-Gilbert/budget_manager/assets/86318284/0a4955fe-3a65-4179-b964-ef0ce07a64d7)

- [x] It is always a beautiful scene when we see tests in projects :clap: . However, I got the below error when I was running ```rspec spec``` and all the tests basically failed.

Screenshot from 2023-07-19 19-04-08

However, when I used ``` bundle exec rspec spec``` the tests then run but with a few failing tests which I will include a screenshot below:

image

image

**In relation to the comment above about the failing linters, kindly do the following:**
- [x] KIndly update your README to include ```bundle exec rspec spec``` as the alternative command to run the tests
- [x] Kindly ensure that all the failing tests are passing correctly 


Cheers, and Happy coding!

File: README.md (hunk 1 above)
Comment (Mutai-Gilbert) [reply README.md:71]: Done

Comment (Mutai-Gilbert) [reply README.md:71]: Done

**Review by Nemwel-Boniface — APPROVED** [review]
##  Hello @Mutai-Gilbert  :wave: 
### This is a manual review
#   :heavy_check_mark: Approve :heavy_check_mark: 

- [x] Thank you for making the changes I had requested :rocket: 



All the tests pass here :partying_face: 

image


Cheers, and Happy coding!

**Comment by bellom** [comment]
@Mutai-Gilbert - kindly go ahead and merge!

**Pull request merged**